### PR TITLE
Textinput runtime barshat

### PR DESCRIPTION
**Pull request by barshat7**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/site/js/textinputview.js [hunk 1]
```diff
@@ -120,6 +122,46 @@
...
+        }
+
+        _runRequiredValidation(widget, model, validationObj) {
+            if (model.required && widget.value === '') {
+                validationObj.isError = true;
+                validationObj.errorMessage = model.constraintMessages['required'];
+            }
+        }
+
+        _runMinMaxValidation(widget, model, validationObj) {
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/site/js/textinputview.js:143]: this should be done by model, not view. view should just act on model reporting some invalidation in subscribe API.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/site/js/textinputview.js [hunk 2]
```diff
@@ -120,6 +122,46 @@
...
+            let validationObj = {isError: false, errorMessage: ''};
+            this._runRequiredValidation(widget, model, validationObj);
+            this._runMinMaxValidation(widget, model, validationObj);
+            this._toggleErrorMark(validationObj);
+        }
+
+        _runRequiredValidation(widget, model, validationObj) {
+            if (model.required && widget.value === '') {
+                validationObj.isError = true;
+                validationObj.errorMessage = model.constraintMessages['required'];
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/site/js/textinputview.js:139]: this is not recommended. pls go by `getState` API in subscribe callback.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/site/js/textinputview.js [hunk 3]
```diff
@@ -120,6 +122,46 @@
 
         _updateValue(value) {
             this.widget.value = value;
+            this._runValidations(this.widget);
+        }
+
+        _runValidations(widget) {
+            const model = this._model._jsonModel;
+            let validationObj = {isError: false, errorMessage: ''};
+            this._runRequiredValidation(widget, model, validationObj);
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/site/js/textinputview.js:131]: view should not run validations. model should run them. view should only be responsible for setting value in model.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/site/js/textinputview.js (hunk 1 above)
Comment (barshat7) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/site/js/textinputview.js:143]: Noted. Thanks for such a quick review! :) It helps a lot.

**Pull request closed without merging**